### ISSUE TITLE
Add support for `plugin` config block and plugin options block

### ISSUE THREAD

**Issue by kaidaguerre** [turbot/steampipe#3807]
# tl;dr
- Add a new `plugin` block in spc files
- Steampipe creates a separate plugin process per `plugin` definition
- move `limiter` to be a child of `plugin`, not a top level block
- add `memory_max_mb` as an argument to `plugin`
- add `memory_max_mb` as an argument to `options "general"`
- add `memory_max_mb` as an argument to new `options "plugin"`
- add env vars: `STEAMPIPE_MEMORY_MAX_MB`, `STEAMPIPE_PLUGIN_MEMORY_MAX_MB`

------

# `plugin` blocks

You may optionally define a `plugin` in a `.spc` file.  This allows you to set plugin-level options.

```hcl
plugin "aws" {
  memory_max_mb = 1000
}
```

The block label is assumed to be the plugin short name unless the `source` argument is present.  The `source` argument accepts the same format as the `connection.plugin` argument. 
These examples all refer to the latest aws plugin:

```hcl
plugin "aws" {}

plugin "my_aws" {
  source = "aws"
}

plugin "my_aws" {
  source = "aws@latest"
}

plugin "my_aws" {
  source = "turbot/aws"
}

plugin "my_aws" {
  source = "hub.steampipe.io/plugins/turbot/aws@latest"
}

```

You can also use specific plugin versions or streams with the `source` argument:
```hcl
# install from major stream
plugin "my_aws" {
  source = "aws@1"
}

# install from minor stream
plugin "my_aws" {
  source = "aws@1.1"
}

# install exact version
plugin "my_aws" {
  source = "aws@0.41.0"
}
```

In a `connection` you may continue to use the current syntax for `plugin` argument
```hcl
connection {
  plugin = "aws"
}
```

-----

# Memory soft limits
2 Environment variables:
  - `STEAMPIPE_MEMORY_MAX_MB`
    - set a soft memory limit for `steampipe` process (no plugins)
    - can also set in general options?
      ```hcl
      options "general" {
        memory_max_mb = 500
        update_check  = true    # true, false
        telemetry     = "info"  # info, none
        log_level     = "info"  # trace, debug, info, warn, error
      }
      ```

    - `STEAMPIPE_PLUGIN_MEMORY_MAX_MB`
      - set a soft memory limit for every plugin that does not set one explicitly
      - can also also set in plugin options??
        ```hcl
        options "plugin" {
          memory_max_mb = 500
        }
        ```
      - a plugin can have its own `memory_max_mb` set in its `plugin` definition.  This value would override the default (`plugin_default_max_memory_mb` / `STEAMPIPE_PLUGIN_DEFAULT_MEMORY_MAX_MB`):
      ```hcl
      plugin "aws" {
        memory_max_mb = 2000
      }
      ```

-----
